catkit/hardware/flir/flir_camera.py

```
@@ -0,0 +1,356 @@
+
+    return property(getter, setter)
+
+
+class FlirCamera(Camera):
+    instrument_lib = FlirLibrary
+
+    _system = None
+    _system_ref_count = 0
+
+    @classmethod
+    def _create_system(cls):
+        '''Create the PySpin system instance and keep ref count for tidy release.
+        '''
+        if cls._system is None:
+            cls._system = cls.instrument_lib.System.GetInstance()
+
+        cls._system_ref_count += 1
+
+    @classmethod
+    def _destroy_system(cls):
+        '''Destroy our reference to the PySpin system instance and release it if there
+        are no references to it anymore.
+        '''
+        cls._system_ref_count -= 1
+
+        if cls._system_ref_count == 0:
+            cls._system.ReleaseInstance()
+            cls._system = None
+
+    def initialize(self):
+        self.log = logging.getLogger(__name__)
+
+        # Set up enum dictionaries
+        self.pixel_format_enum = {'mono8': self.instrument_lib.PixelFormat_Mono8,
+                                  'mono12p': self.instrument_lib.PixelFormat_Mono12p,
+                                  'mono16': self.instrument_lib.PixelFormat_Mono16}
+
+        self.adc_bit_depth_enum = {'8bit': self.instrument_lib.AdcBitDepth_Bit8,
+                                   '10bit': self.instrument_lib.AdcBitDepth_Bit10,
+                                   '12bit': self.instrument_lib.AdcBitDepth_Bit12,
+                                   '14bit': self.instrument_lib.AdcBitDepth_Bit14}
+
+    def _open(self):
+        serial_number = CONFIG_INI.get(self.config_id, 'serial_number')
+
+        # Obtain singleton Flir system object
+        self._create_system()
+
+        self.cam_list = self._system.GetCameras()
+        self.cam = self.cam_list.GetBySerial(serial_number)
+
+        try:
+            self.cam.Init()
+        except:
+            raise RuntimeError(f'Error during intialization of {self.config_id}. Is the camera connected?')
```

[revision] Chain these, please.
```python
...
except Exception as error:
  raise RuntimeError("...") from error
```

---

Also, see https://github.com/spacetelescope/catkit/runs/4484824874?check_suite_focus=true#step:5:10

---

Also, no bare except (just adding a note to myself, I know it's part of your comment).